# Import lib

In [139]:
import pandas as pd
import plotly.graph_objects as go 

# Open file


In [140]:
df = pd.read_excel('data/House prices.xlsx', usecols=range(5), index_col='No')
df.head(10)

Rooms  DistMetro  Area  Price
No                               
1       3         12    56   9396
2       2          4    54   9785
3       1          2    31   5783
4       2          8    48   9123
5       1         11    31   5662
6       3         19    65  11318
7       3         13    69  11301
8       2         12    52   7908
9       3          3    68  13279
10      2          5    46   8047

In [141]:
print(df.shape) # розмір df

(5000, 4)


# Task 1 
Постройте гистограммы распределения цен для каждой категории (по количеству комнат).
Какие выводы можно сделать на основе полученных гистограмм?

In [142]:
fig = go.Figure()

for room in sorted(df.Rooms.unique()):
    room_df = df.query('Rooms == @room')
    n_bins = (room_df.Price.max() - room_df.Price.min()) // 50
    fig.add_trace(go.Histogram(x=room_df.Price, nbinsx=n_bins, opacity=0.75, name=f'room - {room}'))

fig.update_layout(height=600, width=950, xaxis=dict(range=(0, 19000), dtick=1000, title='Price'),
                 yaxis=dict(dtick=5, title='count'), barmode='overlay', title='Rooms-prices hist (bin = 100$)')
    
fig.show()

## Висновок: згідно отриманих результатів, можна сказати, що на ринку нерухомості є велика кількість однокімнатних квартир. Різниця між к-стю двокімнатних та трикімнатних квартир є незначною, в користь двокімнатних квартир. Є області гістограми, які "накладаються". Тобто існують варіанти, які коштують однаково, але к-сть кімнат різна, особливо така картина спостерігається для 2 та 3 - кімнатних квартир. Зі збільшенням кількості кімнат, зростає і ціна, що є очевидним фактом. На ринку нерухомості існують варіанти продажу квартир з к-стю кімнат n+1 за ціною квартири, що має n кімнат, але такі випадки поодинокі. Для більш детального аналізу потрібно розглядати ринок нерухомості враховуючи усі параметри, а не тільки ціну та к-сть кімнат в квартирі.

# Task2
Используя знакомые вам методы, дайте оценку корреляции цены и площади квартиры отдельно по каждой категории (по количеству комнат). Какие выводы можно сделать из этой оценки?

In [143]:
# Кореляція для однокімнатних квартир
df1 = df.loc[df['Rooms'] == 1]
corr1 = df1["Area"].corr(df1["Price"])
corr1

0.6528312289899472

In [144]:
# Кореляція для двокімнатних квартир
df2 = df.loc[df['Rooms'] == 2]
corr2 = df2["Area"].corr(df2["Price"])
corr2

0.7058155230162512

In [145]:
# Кореляція для трикімнатних квартир
df3 = df.loc[df['Rooms'] == 3]
corr3 = df3["Area"].corr(df3["Price"])
corr3

0.7447234199703191

In [146]:
from plotly.offline import iplot
trace_one_room = go.Scatter(x = df1["Area"], y = df1["Price"], marker = dict(color = "green", size = 12),
   mode = "markers", name = "ROOM1", text = "Area/Price")

layout = go.Layout(
    title = "Точкова діаграма для однокімнатних квартир",
    xaxis_title = "Area",
    yaxis_title = "Price",
    height=1000
)
fig = go.Figure(data = trace_one_room, layout = layout)
iplot(fig)

In [147]:
trace_one_room = go.Scatter(x = df2["Area"], y = df2["Price"], marker = dict(color = "red", size = 12),
   mode = "markers", name = "ROOM2", text = "Area/Price")

layout = go.Layout(
    title = "Точкова діаграма для двокімнатних квартир",
    xaxis_title = "Area",
    yaxis_title = "Price",
    height=1000
)
fig = go.Figure(data = trace_one_room, layout = layout)
iplot(fig)

In [148]:
trace_one_room = go.Scatter(x = df3["Area"], y = df3["Price"], marker = dict(color = "gold", size = 12),
   mode = "markers", name = "ROOM3", text = "Area/Price")

layout = go.Layout(
    title = "Точкова діаграма для трикімнатних квартир",
    xaxis_title = "Area",
    yaxis_title = "Price",
    height=1000
)
fig = go.Figure(data = trace_one_room, layout = layout)
iplot(fig)

## Висновок: кореляція у трьої випадках є досить великою та міститься в діапазоні [0,65;0,74].  Тобто можна сказати, що площа квартири має велику залежність від ціни. Коефіцієн кореляції, зі збільшенням к-сті кімнать приблизно збільшується на сталу, а саме на 0,5. Тобто, зі збільшенням к-сті кімнат, збільшується залежність між площею та ціною. За допомогою графічної ілюстрації, а саме побудови точкових діаграм це можна побачити (чим більша площа, тим більша ціна).  Легко визначити мінімальну та максимальну площу для квартир. n = 1 : (21,35); n = 2 : (36,59); n = 3 : (51,83).

# Task3
Рассчитайте границы квартилей распределения цен на 2х квартиры площадью более 40 кв. метров.

In [149]:
# Змінили індексацію
len(df2_40)
df2_40.index = range(1,1447)
df2_40

Rooms  DistMetro  Area  Price
1         2          4    54   9785
2         2          8    48   9123
3         2         12    52   7908
4         2          5    46   8047
5         2          0    49  10305
...     ...        ...   ...    ...
1442      2          6    49   9646
1443      2         19    43   9181
1444      2         18    59  11192
1445      2          0    52   8455
1446      2          6    43   8823

[1446 rows x 4 columns]

In [150]:
df2_40 = df2.query("Area>40")
quartile = pd.DataFrame(df2_40.Price.quantile([0.25,0.50,0.75]))
quartile.index = [1,2,3]
quartile.index.name = 'quartile'
quartile

Price
quartile        
1         7924.0
2         8812.0
3         9684.5

## Висновок : знайшли ціни, що ділять весь ціновий сегмент на чотири рівні частини. Тому двокімнатні квартири з площею більше 40 кв.м. : перші 25% - 7924.0, 50% - 8812.0, 75% - 9684.5 Медіана, тобто квартиль 0,50 - 8812 - середня ціна на квартири. 

# Task4
Вы являетесь риелтором, ваша прибыль это комиссия 3% от суммы сделки купли-продажи квартир (цена квартиры). В текущем квартале у вас ожидается 25 сделок купли-продажи 3-х комнатных квартир. Оцените вероятность того, что ваша суммарная прибыль от этих продаж превысит 9 млн. 200 тыс.

*В каждой категории разная степень корреляции. Т.е. у 3 комнатных она может быть иная, чем у однокомнатных.

# Task5
Рассчитать дисперсию стоимости кв.метра в каждой категории. Как меняется дисперсия с количеством комнат? Есть ли (и какая) связь с изменением коэффициента корреляции?